### PR TITLE
Admin add category

### PR DESCRIPTION
**Pull request by devtuananh**
https://dev.framgia.com/issues/184369

### CODE REVIEW

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -36,6 +36,14 @@ def current_user? user
     user == self
   end
 
+  def admin?
```
Comment (huongnguyenmta) [inline app/models/user.rb]: em dùng `enum` để define role cho người dùng nhé

**Comment by huongnguyenmta** [comment]
commented
I have some comments for this pull :)

File: app/models/user.rb (hunk 1 above)
Comment (devtuananh) [reply app/models/user.rb]: Ok c, 
done

**Comment by devtuananh** [comment]
ready

File: app/models/user.rb [hunk 2]
```diff
@@ -1,5 +1,10 @@
 class User < ApplicationRecord
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  enum role: {admin: Settings.roles.admin,
+    staff: Settings.roles.staff,
+    customer: Settings.roles.customer,
+    locked: Settings.roles.locked
+  }
```
Comment (mdmahmudurrahman) [inline app/models/user.rb]: enum role: [:admin, :staff, customer, :locked]

**Comment by mdmahmudurrahman** [comment]
commented

File: app/models/user.rb (hunk 2 above)
Comment (devtuananh) [reply app/models/user.rb]: Yes, I edited
Thanh you/

**Comment by devtuananh** [comment]
commented

**Comment by devtuananh** [comment]
ready

File: config/settings.yml [hunk 3]
```diff
@@ -5,3 +5,10 @@ validates:
 remember_me_checked: 1
 gravatar_size: 80
 expired_time: 2
+name_category_maximum: 50
+description_minimum: 10
+roles:
+  admin: 2
+  staff: 1
+  customer: 0
```
Comment (mdmahmudurrahman) [inline config/settings.yml]: Now do you need those constants (?)

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by devtuananh** [comment]
commented

**Comment by devtuananh** [comment]
ready

File: config/settings.yml (hunk 3 above)
Comment (devtuananh) [reply config/settings.yml]: Yes, I edited I forgot delete it

**Pull request merged**